src/test/scala/org/broadinstitute/dsde/firecloud/dataaccess/HttpGoogleServicesDAOSpec.scala

```diff
@@ -134,4 +144,54 @@ class HttpGoogleServicesDAOSpec extends AnyFlatSpec with Matchers with PrivateMe
     assertResult(expected) { objectMetadata }
 
   }
+
+  it should "return GcsPath for a successful object upload" in {
+    // create local storage service
+    val db = LocalStorageHelper.getOptions().getService()
+    val localStorage = storageResource(db)
+
+    val bucketName = GcsBucketName("some-bucket")
+    val objectName = GcsObjectName("folder/object.json")
+
+    val objectContents = "Hello world".getBytes(StandardCharsets.UTF_8)
+
+    assertResult(GcsPath(bucketName, objectName)) {
+      gcsDAO.streamUploadObject(localStorage, bucketName, objectName, objectContents)
+    }
+  }
+
+  it should "throw error for an unsuccessful object upload" in {
```

Comment:
This test feels like we we're just verifying that if we mock Storage to throw an exception that it does in fact throw an exception, but it's possible that I'm missing something something, i.e. testing the mock framework rather than our code.

Reply:
The intention of this test is to ensure that if Storage throws an exception, we bubble that exception up through the other layers inside `streamUploadObject`. I admit there's only a thin additional layer, but I'd like to keep the test, unless you object